### PR TITLE
add MarquezWriteOnlyClient

### PR DESCRIPTION
**Pull request by julienledem**
See: https://github.com/MarquezProject/marquez/issues/798
Signed-off-by: Julien Le Dem <julien@apache.org>

### CODE REVIEW

**Review by sreev — APPROVED** [review]
awesome.

File: src/main/java/marquez/client/MarquezWriteOnlyClientImpl.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        break;
+      case FAILED:
+        transition = "fail";
+        break;
+      default:
+        throw new IllegalArgumentException(
+            String.format("Unexpected run state: %s", runState.name()));
+    }
+    backend.post(
+        path("/jobs/runs/%s/%s", ImmutableMap.of("at", ISO_INSTANT.format(at)), runId, transition));
```
Comment (wslulciuc) [inline src/main/java/marquez/client/MarquezWriteOnlyClientImpl.java:105]: _Minor_: Given that paths will be shared between the _standard_ client and _write only_ client, we may want to think of a way to reduce duplication. Not to be addressed in this PR, but something to think about.

**Comment by wslulciuc** [comment]
Great work, @julienledem! I think we'll need to update [`RunMeta`](https://github.com/MarquezProject/marquez-java/blob/feature/write-only-client/src/main/java/marquez/client/models/RunMeta.java) to support an external run ID?

File: src/main/java/marquez/client/MarquezWriteOnlyClientImpl.java (hunk 1 above)
Comment (julienledem) [reply src/main/java/marquez/client/MarquezWriteOnlyClientImpl.java:105]: Agreed, I didn't want to touch the existing client too much in this PR. I would rework the MarquezUrl to share more logic.

**Comment by julienledem** [comment]
> Great work, @julienledem! I think we'll need to update [`RunMeta`](https://github.com/MarquezProject/marquez-java/blob/feature/write-only-client/src/main/java/marquez/client/models/RunMeta.java) to support an external run ID?

good point!

**Pull request merged**